orthoinference/Jenkinsfile

```diff
@@ -0,0 +1,107 @@
+				// It takes all Human Reaction instances in the database and attempts to project each Reaction to each species by
+				// stripping them down to the reactio's constituent proteins, checks if the protein homolog exists for that species, and infers it in Reactome's data model.
+				// If enough proteins (>= 75%) are inferrable in a Reaction, then it is created and stored in the database for this release. This is done from scratch each time.
+				script{
+					speciesList = ['mmus', 'rnor', 'cfam', 'btau', 'sscr', 'drer', 'xtro', 'ggal', 'dmel', 'cele', 'ddis', 'spom', 'scer', 'pfal']
+					for (species in speciesList) {
+						stage("Main: Infer ${species}"){
+							script{
+								dir('orthoinference'){
+									withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]){
+										sh "java -Xmx${env.JAVA_MEM_MAX}m -jar target/orthoinference-${env.ORTHOINFERENCE_VERSION}-jar-with-dependencies.jar $ConfigFile ${species}"
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+		// This stage backs up the release_current database after it is modified.
+		stage('Post: Backup DB'){
+			steps{
+				script{
+					dir('orthoinference'){
+						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							def release_current_after_orthoinference_dump = "${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_after_orthoinference.dump"
+							sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > $release_current_after_orthoinference_dump"
+							sh "gzip -f $release_current_after_orthoinference_dump"
+						}
+					}
+				}
+			}
+		}
+		// This stage archives all logs and database backups produced by Orthoinference. It also archives the eligible/inferred files produced by orthoinference.
+		stage('Archive logs and backups'){
+			steps{
+				script{
+					dir('orthoinference'){
+						sh "mkdir -p archive/${env.RELEASE_NUMBER}/logs"
+						sh "mv --backup=numbered *_${env.RELEASE_NUMBER}_*.dump.gz archive/${env.RELEASE_NUMBER}/"
+						sh "gzip logs/*"
+						sh "mv logs/* archive/${env.RELEASE_NUMBER}/logs/"
+						sh "mkdir -p ${env.RELEASE_NUMBER}"
+						sh "gzip -f *.txt"
+						sh "mv *.txt.gz ${env.RELEASE_NUMBER}/"
+					}
+				}
+			}
+		}
+	}
+}
+
+// Utility function that checks upstream builds of this project were successfully built.
+def checkUpstreamBuildsSucceeded(String stepName, String currentRelease) {
+	def statusUrl = httpRequest authentication: 'jenkinsKey', validResponseCodes: '200:404', url: "${env.JENKINS_JOB_URL}/job/$currentRelease/job/$stepName/lastBuild/api/json"
+	if (statusUrl.getStatus() == 404) {
+		error("$stepName has not yet been run. Please complete a successful build.")
+	} else {
+		def statusJson = new JsonSlurper().parseText(statusUrl.getContent())
+		if(statusJson['result'] != "SUCCESS"){
+			error("Most recent $stepName build status: " + statusJson['result'] + ". Please complete a successful build.")
+		}
+	}
+}
```

Comment:
This is good!  

Can it be moved to a higher level so every step can access it?  It would help remove duplicated logic between other steps.